addons/pos_self_order/static/src/kiosk/components/attribute_selection/attribute_selection.js

```diff
@@ -38,29 +38,66 @@ export class AttributeSelection extends Component {
         }
     }
 
-    attributeClicked() {
-        if (this.state.editMode) {
+    attributeClicked(attrId) {
+        const curAttr = this.attribute;
+
+        if (curAttr && curAttr.display_type === "multi") {
+            if (!this.props.state.selectedVariants[curAttr.id]) {
+                this.props.state.selectedVariants[curAttr.id] = new Set();
+            }
+
+            if (this.props.state.selectedVariants[curAttr.id].has(attrId)) {
+                this.props.state.selectedVariants[curAttr.id].delete(attrId);
+            } else {
+                this.props.state.selectedVariants[curAttr.id].add(attrId);
+            }
+        }
+
+        if (this.state.editMode && curAttr.display_type !== "multi") {
             this.state.editMode = false;
             this.state.showResume = true;
-            this.props.productState.showQtyButtons = true;
+            this.props.state.showQtyButtons = true;
             return;
         }
-        this.next();
+
+        if (curAttr && curAttr.display_type !== "multi") {
+            this.next();
+        }
+    }
+
+    isChecked(attribute, value) {
+        return (
+            this.props.state.selectedVariants[attribute.id] instanceof Set &&
+            this.props.state.selectedVariants[attribute.id].has(parseInt(value.id))
+        );
```

Comment:
this would be so much easier if `selectedVariants[attribute.id]` was simply an object with `value.id` as key and and a bool as value. Then you could use t-model and not need the ` attributeClicked(attrId)` handler

Reply:
We already talk about that.. Look above